src/Uno.UWP/UI/Colors.cs

```diff
@@ -71,45 +71,82 @@ public static Color Parse(string colorCode)
 		public static Color FromARGB(string colorCode)
 		{
 			byte a, r, b, g;
+			int offset;
+			int len;
 
-			colorCode = colorCode.TrimStart(new char[] { '#' });
+			if (colorCode is null)
+			{
+				len = 0;
+				offset = 0;
+			}
+			else
+			{
+				len = colorCode.Length;
+				// skip a starting `#` if present 
+				offset = (len > 0 && colorCode[0] == '#' ? 1 : 0);
+				len -= offset;
+			}
 
-			if (colorCode.Length == 3)
+			// deal with an optional alpha value
+			if (len == 4)
 			{
-				a = 0xFF;
-				r = Convert.ToByte(new String(colorCode[0], 2), 16);
-				g = Convert.ToByte(new String(colorCode[1], 2), 16);
-				b = Convert.ToByte(new String(colorCode[2], 2), 16);
+				a = ToByte(colorCode[offset++]);
+				a = (byte)(a << 4 + a);
+				len = 3;
 			}
-			else if (colorCode.Length == 4)
+			else if (len == 8)
 			{
-				a = Convert.ToByte(new String(colorCode[0], 2), 16);
-				r = Convert.ToByte(new String(colorCode[1], 2), 16);
-				g = Convert.ToByte(new String(colorCode[2], 2), 16);
-				b = Convert.ToByte(new String(colorCode[3], 2), 16);
+				a = (byte)((ToByte(colorCode[offset++]) << 4) + ToByte(colorCode[offset++]));
+				len = 6;
 			}
-			else if (colorCode.Length == 6)
+			else
 			{
 				a = 0xFF;
-				r = Convert.ToByte(colorCode.Substring(0, 2), 16);
-				g = Convert.ToByte(colorCode.Substring(2, 2), 16);
-				b = Convert.ToByte(colorCode.Substring(4, 2), 16);
 			}
-			else if (colorCode.Length == 8)
+
+			// the process the required R G and B values
+			if (len == 3)
+			{
+				r = ToByte(colorCode[offset++]);
+				r = (byte)(r << 4 + r);
+				g = ToByte(colorCode[offset++]);
+				g = (byte)(g << 4 + g);
+				b = ToByte(colorCode[offset++]);
+				b = (byte)(b << 4 + b);
+			}
+			else if (len == 6)
 			{
-				a = Convert.ToByte(colorCode.Substring(0, 2), 16);
-				r = Convert.ToByte(colorCode.Substring(2, 2), 16);
-				g = Convert.ToByte(colorCode.Substring(4, 2), 16);
-				b = Convert.ToByte(colorCode.Substring(6, 2), 16);
+				r = (byte)((ToByte(colorCode[offset++]) << 4) + ToByte(colorCode[offset++]));
+				g = (byte)((ToByte(colorCode[offset++]) << 4) + ToByte(colorCode[offset++]));
+				b = (byte)((ToByte(colorCode[offset++]) << 4) + ToByte(colorCode[offset++]));
 			}
 			else
 			{
-				a = 0xFF;
 				r = 0xFF;
 				g = 0x0;
 				b = 0x0;
```

Comment:
This looks like an error code path? I guess we probably should be throwing here instead of returning "Red" randomly here. This can be in another PR since this PR is focused on refactoring to improve performance without behavioral changes.

Reply:
Yes, that's the original behaviour.

From previous experience (silverlight/moonlight) using a color was more useful than an exception to debug color issues, since an exception was very hard to map back to the code that generated the exception. Not sure if this applies here... but that can be discussed on an issue/PR :)